### PR TITLE
Step 7 with Tests

### PR DESCRIPTION
**Pull request by gabinas**
LoginServlet and LoginServletTest modified for P1 Step 7

### CODE REVIEW

File: src/main/java/codeu/controller/LoginServlet.java [hunk 1]
```diff
@@ -58,28 +58,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+ /**
+  * This function fires when a user submits the login form. It gets the username and password from
+  * the submitted form data, checks that they're valid, and either adds the user to the session
+  * so we know the user is logged in or shows an error to the user.
+  */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+	String password = request.getParameter("password");
```
Comment (rcagle) [inline src/main/java/codeu/controller/LoginServlet.java]: Nit: horizontal alignment is off.

File: src/test/java/codeu/controller/LoginServletTest.java [hunk 2]
```diff
@@ -51,56 +51,70 @@ public void testDoGet() throws IOException, ServletException {
...
-    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
-
-    loginServlet.doPost(mockRequest, mockResponse);
-
-    Mockito.verify(mockRequest)
-        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
-    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
-  }
-
+
```
Comment (rcagle) [inline src/test/java/codeu/controller/LoginServletTest.java]: Nit: just leave 1 blank line

File: src/main/java/codeu/controller/LoginServlet.java (hunk 1 above)
Comment (rcagle) [reply src/main/java/codeu/controller/LoginServlet.java]: That applies to the tests, too.

File: src/test/java/codeu/controller/LoginServletTest.java [hunk 3]
```diff
@@ -51,56 +51,70 @@ public void testDoGet() throws IOException, ServletException {
...
     Mockito.verify(mockResponse).sendRedirect("/conversations");
   }
+  
+  
+  @Test
+  public void testDoPost_NewUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+	Mockito.when(mockRequest.getParameter("password")).thenReturn("testPassword");
 
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
```
Comment (rcagle) [inline src/test/java/codeu/controller/LoginServletTest.java]: Make your mocks member variables and initialize them in an @Setup method.

File: src/main/java/codeu/controller/LoginServlet.java [hunk 4]
```diff
@@ -58,28 +58,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
-
-    request.getSession().setAttribute("user", username);
-    response.sendRedirect("/conversations");
+    if (userStore.isUserRegistered(username)) {
+      User user = userStore.getUser(username);
+	  if(password.equals(user.getPassword())){
+		  request.getSession().setAttribute("user",username);
+		  response.sendRedirect("/conversations");
+	  }
+	  else{
```
Comment (rcagle) [inline src/main/java/codeu/controller/LoginServlet.java]: Nit: can you please follow the formatting used in the rest of these files?

if () {
} else {
}

**Review by rcagle — COMMENTED** [review]
I'd like to re-review this after you've addressed my comments.

**Comment by gabinas** [comment]
I've fixed all things as indicated!  @rcagle

File: src/main/java/codeu/controller/LoginServlet.java [hunk 5]
```diff
@@ -58,28 +58,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
 
-    if (!username.matches("[\\w*\\s*]*")) {
-      request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
-      request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
-      return;
+    if (userStore.isUserRegistered(username)) {
+      User user = userStore.getUser(username);
+      if(password.equals(user.getPassword())){
```
Comment (rcagle) [inline src/main/java/codeu/controller/LoginServlet.java]: Nit: this if statement doesn't have a space in front of it. (Applies throughout)

**Review by rcagle — DISMISSED** [review]
Could you please fix the spaces on your if and else statements? Other than that, LGTM.

**Comment by gabinas** [comment]
Fixed !

**Review by rcagle — APPROVED** [review]

**Pull request merged**